openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -438,6 +440,37 @@ def test_course_goals(self):
         response = self.client.get(course_home_url(audit_only_course))
         self.assertNotContains(response, TEST_COURSE_GOAL_OPTIONS)
 
+    @override_waffle_flag(UNIFIED_COURSE_TAB_FLAG, active=True)
+    @override_waffle_flag(COURSE_PRE_START_ACCESS_FLAG, active=True)
+    @override_waffle_flag(ENABLE_COURSE_GOALS, active=True)
+    def test_course_goal_updates(self):
+        """
+        Ensure that the following five use cases work as expected.
+
+        1) Unenrolled users are not shown the update goal selection field.
+        2) Enrolled users are not shown the update goal selection field if they have not yet set a course goal.
+        3) Enrolled users are shown the update goal selection field if they have set a course goal.
+        """
```

Comment:
Based on earlier comments, I "think" that a user that has already verified will not be able to see the update and they should?  Do we need to handle the auto-goal setting?  Where did we land on that?

Reply:
Do you want to add the following above?
```
4) Enrolled users in the verified track are shown the update goal selection field.
```